lib/components/parallax_component.dart

```diff
@@ -196,18 +200,14 @@ class ParallaxComponent extends PositionComponent {
   @mustCallSuper
   @override
   void render(Canvas canvas) {
-    if (!loaded) {
-      return;
-    }
     super.render(canvas);
     canvas.save();
     _layers.forEach((layer) => layer.render(canvas));
     canvas.restore();
   }
 
-  void _load(List<ParallaxImage> images) {
+  Future<void> _load(List<ParallaxImage> images) async {
     _layers = images.map((image) => ParallaxLayer(image)).toList();
-    Future.wait(_layers.map((layer) => layer.future))
-        .then((_) => loaded = true);
+    await Future.wait(_layers.map((layer) => layer.future));
```

Comment:
Same here, we don't want to be blocking right?

Reply:
Same answer from the child method